### PR TITLE
Update release-toolkit to 2.3.0

### PR DESCRIPTION
**Pull request by oguzkocer**
This PR is a smaller subset of https://github.com/wordpress-mobile/WordPress-iOS/pull/17626. We are making this change in the release branch as well to avoid any issues with the default branch being updated.

**Please DO NOT merge this PR until the default branch change is applied.**

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

**PR submission checklist:**

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: fastlane/Pluginfile [hunk 1]
```diff
@@ -6,9 +6,9 @@ group :screenshots, optional: true do
   gem 'rmagick', '~> 3.2.0'
 end
 
-# gem 'fastlane-plugin-wpmreleasetoolkit', '~> 2.1'
+gem 'fastlane-plugin-wpmreleasetoolkit', '~> 2.3'
 # This comment avoids typing to switch to a development version for testing.
-gem 'fastlane-plugin-wpmreleasetoolkit', git: 'git@github.com:wordpress-mobile/release-toolkit', branch: 'develop'
+#gem 'fastlane-plugin-wpmreleasetoolkit', git: 'git@github.com:wordpress-mobile/release-toolkit', branch: 'develop'
```
Comment (AliSoftware) [inline fastlane/Pluginfile]:
```suggestion
#gem 'fastlane-plugin-wpmreleasetoolkit', git: 'git@github.com:wordpress-mobile/release-toolkit', branch: 'trunk'
```

**Comment by jkmassel** [comment]
Closing this one, because the release toolkit version was updated as part of https://github.com/wordpress-mobile/WordPress-iOS/pull/17683.

**Comment by mokagio** [comment]
Thanks @jkmassel, I had missed this!

**Comment by oguzkocer** [comment]
@jkmassel This was still a relevant PR because it updates the branch to `trunk`. Updating the base branch to the new release branch would have gotten rid of all the other changes. Anyway, I'll open a new one for the branch change 🤷‍♂️

**Pull request closed without merging**